platform/core/src/classes/CommandsManager.ts

```diff
@@ -161,6 +162,49 @@ export class CommandsManager {
       return commandFn(commandParams);
     }
   }
+
+  /**
+   * Run one or more commands with specified extra options.
+   *
+   * @param toRun - A specification of one or more commands
+   * @param extraOptions - to include in the commands run beyond
+   *   the commandOptions specified in the base.
+   */
+  public run(
+    toRun: Command | Commands | Command[] | undefined,
+    extraOptions?: Record<string, unknown>
+  ): unknown {
+    if (!toRun) return;
+    const commands =
+      (Array.isArray(toRun) && toRun) ||
+      ((toRun as Command).commandName && [toRun]) ||
+      (Array.isArray((toRun as Commands).commands) &&
```

Comment:
This doesn't get hit since you have the same `Array.isArray` in the first check

Reply:
Actually it gets hit fairly easily - there are three patterns that are ok:
{ commandName: "....", ... }
[{commandName: "..."},....]
{ commands: [{commandName: "...",...},...] }
The last one is actually a common pattern - eg every tool uses it.